File: src/main/java/com/redbox/domain/donation/entity/DonationGroup.java
package com.redbox.domain.donation.entity;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Getter
@Entity
@Table(name = "donation_groups")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class DonationGroup {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "donation_group_id")
    private Long id;
    private Long donorId;
    private Long receiverId;
    @Enumerated(EnumType.STRING)
    private DonationType donationType;
    private int donationAmount;
    private LocalDate donationDate;
    private String donationMessage;
    @Enumerated(EnumType.STRING)
    private DonationStatus donationStatus;

    @Builder
    public DonationGroup(Long donorId, Long receiverId, DonationType donationType, int donationAmount, LocalDate donationDate, String donationMessage, DonationStatus donationStatus) {
        this.donorId = donorId;
        this.receiverId = receiverId;
        this.donationType = donationType;
        this.donationAmount = donationAmount;
        this.donationDate = donationDate;
        this.donationMessage = donationMessage;
        this.donationStatus = donationStatus;
    }

    public void donateConfirm() {
        this.donationStatus = DonationStatus.DONE;
    }

    public void donateCancel() {
        this.donationStatus = DonationStatus.CANCEL;
    }
}

File: legacy/src/main/java/com/redbox/domain/donation/entity/DonationGroup.kt
package com.redbox.domain.donation.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.time.LocalDate

@Entity
@Table(name = "donation_groups")
class DonationGroup(
    donorId: Long,
    receiverId: Long,
    donationType: DonationType,
    donationAmount: Int,
    donationDate: LocalDate,
    donationMessage: String,
    donationStatus: DonationStatus
) {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "donation_group_id")
    var id: Long? = null
        private set

    @Column(name = "donor_id", nullable = false)
    val donorId: Long = donorId

    @Column(name = "receiver_id", nullable = false)
    val receiverId: Long = receiverId

    @Enumerated(EnumType.STRING)
    @Column(name = "donation_type", nullable = false)
    val donationType: DonationType = donationType

    @Column(name = "donation_amount", nullable = false)
    val donationAmount: Int = donationAmount

    @Column(name = "donation_date", nullable = false)
    val donationDate: LocalDate = donationDate

    @Column(name = "donation_message", nullable = false)
    val donationMessage: String = donationMessage

    @Enumerated(EnumType.STRING)
    @Column(name = "donation_status", nullable = false)
    var donationStatus: DonationStatus = donationStatus

    // 상태 변경 메서드
    fun donateConfirm() {
        this.donationStatus = DonationStatus.DONE
    }

    fun donateCancel() {
        this.donationStatus = DonationStatus.CANCEL
    }
}